src/cmd/sign.js

```diff
@@ -13,12 +13,46 @@ import {createLogger} from '../util/logger';
 const log = createLogger(__filename);
 export const extensionIdFile = '.web-extension-id';
 
-export default function sign(
-    {verbose, sourceDir, artifactsDir, apiKey, apiSecret,
-     apiUrlPrefix, id, timeout}: Object,
-    {build=defaultBuilder, signAddon=defaultAddonSigner,
-     preValidatedManifest=null}: Object = {}): Promise<Object> {
+// Flow Types
+
+import type {ExtensionManifest} from  '../util/manifest';
+
+export type SignCmdParams = {
+  id?: string,
+  verbose?: boolean,
+  sourceDir: string,
+  artifactsDir: string,
+  apiKey: string,
+  apiSecret: string,
+  apiUrlPrefix: string,
+  timeout: number,
+};
+
+export type SignCmdOptions = {
+  build?: typeof defaultBuilder,
+  signAddon?: typeof defaultAddonSigner,
+  preValidatedManifest?: ExtensionManifest,
+};
 
+export type SignCmdResult = {
+  success: boolean,
+  id: string,
+  downloadedFiles: Array<string>,
+};
+
+
+// Module internals & exports
```

Comment:
in this case, we do need to put type declarations below this line but elsewhere, I think there will not be a clean delineation. Maybe this comment marker should be removed everywhere? Just a nit.